lib/internment.rs

```diff
@@ -1,190 +1,281 @@
-use differential_datalog::record;
-use differential_datalog::record::Record;
-use std::cmp;
-use std::fmt;
-use std::hash::Hash;
-
+use arc_interner::ArcIntern;
+use ddlog_std::Vec as DDlogVec;
+use differential_datalog::record::{self, Record};
+use serde::{de::Deserializer, ser::Serializer};
+use std::{
+    cmp::{self, Ordering},
+    fmt::{Debug, Display, Formatter, Result as FmtResult},
+    hash::Hash,
+};
+
+/// An atomically reference counted handle to an interned value
+///
+/// While this type is thread safe, it may cause logical bugs
```

Comment:
Thanks for the PR!

I don't understand this paragraph. How can one rely on a variable being pre-interned? What kind of usage pattern are you trying to prevent here?

Reply:
It's just letting the user know that it's a global store of all interned items

Reply:
How can this lead to logical bugs?

The rest of the PR looks great, but I feel that this comment needs to be removed or rephrased. Perhaps just pointing to the [`arc_interner`](https://crates.io/crates/arc-interner) crate documentation is good enough.